Snowflake.Data.Tests/SFDbCommandIT.cs

```diff
@@ -595,5 +597,705 @@ public void TestRowsAffectedUnload()
+                    DateTime ntz2 = DateTime.ParseExact("2020-12-31 23:59:59", "yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture);
+                    DateTime ntz3 = DateTime.ParseExact("2022-04-01 00:00:00", "yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture);
+                    List<DateTime> arrNtz = new List<DateTime>();
+                    for (int i = 0; i < total; i++)
+                    {
+                        arrNtz.Add(ntz1);
+                        arrNtz.Add(ntz2);
+                        arrNtz.Add(ntz3);
+                    }
+                    var p5 = cmd.CreateParameter();
+                    p5.ParameterName = "5";
+                    p5.DbType = DbType.DateTime2;
+                    p5.Value = arrNtz.ToArray();
+                    cmd.Parameters.Add(p5);
+
+                    DateTimeOffset tz1 = DateTimeOffset.Now;
+                    DateTimeOffset tz2 = DateTimeOffset.UtcNow;
+                    DateTimeOffset tz3 = new DateTimeOffset(2007, 1, 1, 12, 0, 0, new TimeSpan(4, 0, 0));
+                    List<DateTimeOffset> arrTz = new List<DateTimeOffset>();
+                    for (int i = 0; i < total; i++)
+                    {
+                        arrTz.Add(tz1);
+                        arrTz.Add(tz2);
+                        arrTz.Add(tz3);
+                    }
+
+                    var p6 = cmd.CreateParameter();
+                    p6.ParameterName = "6";
+                    p6.DbType = DbType.DateTimeOffset;
+                    p6.Value = arrTz.ToArray();
+                    cmd.Parameters.Add(p6);
+
+                    Task<int> task = cmd.ExecuteNonQueryAsync(externalCancel.Token);
+
+                    task.Wait();
+                    Assert.AreEqual(total * 3, task.Result);
+
+                    cmd.CommandText = "SELECT * FROM testPutArrayBind";
+                    IDataReader reader = cmd.ExecuteReader();
+                    Assert.IsTrue(reader.Read());
+
+                    //cmd.CommandText = "drop table if exists testPutArrayBind";
+                    //cmd.ExecuteNonQuery();
+
+                }
+
+                conn.Close();
+            }
+        }
+
+        [Test]
+        public void TestPutArrayBindAsyncMultiThreading()
+        {
+            Thread t1 = new Thread(() => ThreadProcess1(ConnectionString));
+            Thread t2 = new Thread(() => ThreadProcess2(ConnectionString));
+            //Thread t3 = new Thread(() => ThreadProcess3(ConnectionString));
+            //Thread t4 = new Thread(() => ThreadProcess4(ConnectionString));
+
+            t1.Start();
+            t2.Start();
+            //t3.Start();
+            //t4.Start();
+        }
+
+        static void ThreadProcess1(string connstr)
+        {
+            CancellationTokenSource externalCancel = new CancellationTokenSource(TimeSpan.FromSeconds(100));

```

Comment:
This looks like exact same as testPutArrayBindAsync(). Can we reduce duplicate of code?